src/Core.zig

```diff
@@ -183,14 +183,15 @@ fn init(core: *Mod, entities: *mach.Entities.Mod) !void {
     var events = EventQueue.init(allocator);
     try events.ensureTotalCapacity(8192);
 
+    const platform: Platform = undefined;
     core.init(.{
         .allocator = allocator,
         .main_window = main_window,
         .events = events,
         .input_state = .{},
 
         // TODO: remove undefined initialization (disgusting!)
-        .platform = undefined,
```

Comment:
Had to do this because Zig was unsure of the type of `platform` during compile time, and it needed to know that before passing it into `try Platform.init()` [here](https://github.com/hexops/mach/pull/1268/files#diff-5d71102262b2ae72ea5d47ecbde60a14e74dfea97b9351dd2610021f39b88774L209)